src/game/Entities/Unit.cpp

```diff
@@ -3271,7 +3271,7 @@ bool Unit::CanGlance() const
 bool Unit::CanDaze() const
 {
     // Generally, only npcs are able to daze targets in melee
-    return (GetTypeId() == TYPEID_UNIT && !HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PLAYER_CONTROLLED));
+    return (GetTypeId() == TYPEID_UNIT && !HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PLAYER_CONTROLLED) && !static_cast<const Creature*>(this)->GetSettings().HasFlag(CreatureStaticFlags4::CANNOT_DAZE));
```

Comment:
Make the function virtual, return false by default, override it in creature and there do isplayercontrolled and the other check (at this point we know how daze works so encapsulation wont hinder anything and will be nicer)

Reply:
done